### PR TITLE
Export : Standalone API Resource

### PR DESCRIPTION
**Pull request by moizarafat**
## Description
API Resource for Standalone Module.

## Motivation and Context
Supports Data Export.

## How Has This Been Tested?
Existing and New test cases.

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: elide-async/src/test/java/com/yahoo/elide/async/resources/ExportApiEndpointTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        String queryId = "1";
+        when(engine.getResultsByID(queryId)).thenReturn(Observable.just("result"));
+
+        exportApiProperties = new ExportApiProperties(Executors.newFixedThreadPool(1), 10);
+        endpoint = new ExportApiEndpoint(engine, exportApiProperties);
+        endpoint.get(queryId, response, asyncResponse);
+
+        verify(engine, Mockito.times(1)).getResultsByID(queryId);
+        // Sleep to let the request finish.
+        Thread.sleep(100000);
```
Comment (aklish) [inline elide-async/src/test/java/com/yahoo/elide/async/resources/ExportApiEndpointTest.java:63]: We really don't want to add sleep here in the tests.   This slows down the builds and is brittle.

File: elide-standalone/src/main/java/com/yahoo/elide/standalone/config/ElideResourceConfig.java [hunk 2]
```diff
@@ -151,8 +153,19 @@ protected void configure() {
                     }
                     bind(asyncAPIDao).to(AsyncAPIDAO.class);
 
-                    // TODO: If null, initialize with FileResultStorageEngine
                     ResultStorageEngine resultStorageEngine = asyncProperties.getResultStorageEngine();
+                    if (resultStorageEngine == null) {
```
Comment (aklish) [inline elide-standalone/src/main/java/com/yahoo/elide/standalone/config/ElideResourceConfig.java:157]: Why don't we move the initialization of ResultStorageEngine inside the if block for export enabled?

File: elide-standalone/src/test/java/example/ElideStandaloneExportTest.java [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+/**
+ * Tests ElideStandalone starts and works.
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ElideStandaloneExportTest {
+    protected ElideStandalone elide;
+
+    @BeforeAll
+    public void init() throws Exception {
+        elide = new ElideStandalone(new ElideStandaloneSettings() {
```
Comment (aklish) [inline elide-standalone/src/test/java/example/ElideStandaloneExportTest.java:43]: Most of this is boilerplate.  Let's make a new test class that implements ElideStandaloneSettings (ElideStandaloneTestSettings) - that we can parameterized and remove this boilerplate.

**Review by aklish — CHANGES_REQUESTED** [review]

File: elide-async/pom.xml [hunk 4]
```diff
@@ -90,6 +90,13 @@
             <version>2.4.0</version>
         </dependency>
 
+        <dependency>
+            <groupId>javax.servlet</groupId>
```
[the remark this answers is not in the document]
Comment (moizarafat) [reply elide-async/pom.xml:94]: Moving the Resource class to standalone,

File: elide-async/src/test/java/com/yahoo/elide/async/resources/ExportApiEndpointTest.java (hunk 1 above)
Comment (moizarafat) [reply elide-async/src/test/java/com/yahoo/elide/async/resources/ExportApiEndpointTest.java:63]: Replaced Thread.sleep with Junit After.

File: elide-standalone/src/main/java/com/yahoo/elide/standalone/config/ElideResourceConfig.java [hunk 5]
```diff
@@ -159,17 +159,19 @@ protected void configure() {
                     }
                     bind(asyncAPIDao).to(AsyncAPIDAO.class);
 
-                    ResultStorageEngine resultStorageEngine = asyncProperties.getResultStorageEngine();
-                    if (resultStorageEngine == null) {
-                        resultStorageEngine = new FileResultStorageEngine(asyncProperties.getStorageDestination());
-                    }
-                    bind(resultStorageEngine).to(ResultStorageEngine.class).named("resultStorageEngine");
+                    ResultStorageEngine resultStorageEngine = null;
```
Comment (aklish) [inline elide-standalone/src/main/java/com/yahoo/elide/standalone/config/ElideResourceConfig.java:162]: Why not move this inside the if block too?

File: elide-standalone/src/test/java/example/ElideStandaloneTestSettings.java [hunk 6]
```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2018, Oath Inc.
```
Comment (aklish) [inline elide-standalone/src/test/java/example/ElideStandaloneTestSettings.java:2]: 2021 Yahoo

**Review by aklish — CHANGES_REQUESTED** [review]

**Review by aklish — APPROVED** [review]

**Pull request merged**